growth/too/tasks/tiles.py

```diff
@@ -301,6 +300,24 @@ def get_planned_observations(
                         corners[2] = corners_copy[3]
                         corners[3] = corners_copy[2]
 
+                    fields = models.Field.query.filter_by(telescope=telescope,
+                                                          ra=ra, dec=dec).all()
+                    if len(fields) == 0:
+                        contour = {}
+                        field = models.Field(telescope=telescope,
+                                             ra=ra, dec=dec,
+                                             contour=contour,
+                                             reference_filter_ids=[],
+                                             reference_filter_mags=[],
+                                             ipix=ipix.tolist())
+                        models.db.session.merge(field)
+
+                        fields = models.Field.query.filter_by(
+                            telescope=telescope, ra=ra, dec=dec).all()
+
+                    field = fields[0]
+                    field_maps[field_id] = field.field_id
+
                     contour = {
```

Comment:
Please move all of this up to where you currently have `contours = {}` so that we only have to do one database transaction.